File: js/day2-longest-string.js
/**
 * 100 Days Coding Challenge
 * 
 * @author Rhan Miano
 * @since November 6, 2018
 * 
 * Pushing myself to do this 100 Days Coding Challenge.
 */

// Day 2 Find the longest word in a string

/**
 * Split the words with [space] delimiter
 * and loop through it. For each loop create an
 * object that will hold the word itself and its
 * length, the object will then be pushed 
 * to arrTemp[] and gets sorted.
 * Returns an array.
 */
let getLongestString = function (args){
  try{
    let arrWords = args.split(" "),
        arrTemp = [];
        arrLongestWord = [];
    arrWords.forEach(function(value){
      let objValues    = {};   
      objValues.word   = value;
      objValues.length = value.length;

      arrTemp.push(objValues);
    });
    
    arrTemp.sort(function(a,b){
      return a.length - b.length;
    });

    arrLongestWord.push(arrTemp[arrTemp.length - 1]);

    // What if there are other words with equal lengths?
    let arrWordsSameLength = [];
    arrTemp.forEach(function(value){
      if(arrLongestWord[0].length == value.length) {
        arrWordsSameLength.push(value);
      }
    });

    // Returns an array of word or words with the longest length
    return arrWordsSameLength.length > 0 ? arrWordsSameLength : arrLongestWord;
  } catch(err){
    return `${err.name}: Given param is not a string and therefore ${err.message}`;
  }  
}

File: py/day2-longest-string.py
##
 # 100 Days Coding Challenge
 # 
 # @author Rhan Miano
 # @since November 6, 2018
 # 
 # Pushing myself to do this 100 Days Coding Challenge.
##

# Day 2 Find the longest word in a string
#   Find the length of the longest word in the provided sentence. 
#   Your response should be the longest word and its length.

def getLongestString(args):
  arrWords = args.split(" ")  
  arrWords.sort(key=getLength)

  longestWord = arrWords[len(arrWords) - 1]

  # What if there are other words with equal lengths?
  arrWordsSameLength = []
  for word in arrWords:
    if(len(longestWord) == len(word)):
      arrWordsSameLength.append(word)
  
  if(len(arrWordsSameLength) > 0):
    print("Longest Words: {}".format(arrWordsSameLength))
    print("Length: {}".format(len(arrWordsSameLength[0])))
  else:
    print("Longest Word: {}".format(longestWord))
    print("Length: {}".format(len(longestWord[0])))

def getLength(args):
  return len(args)

getLongestString("hi how are you")